tests/unit/linalg/test_matrix_rectangular.cpp

```diff
@@ -49,43 +49,60 @@ TEST_CASE("FormRectangular", "[FormRectangularSystemMatrix]")
       // Scalar
       H1_FECollection fec1(order, dim);
       FiniteElementSpace fes1(&mesh, &fec1);
-
-      // Vector valued
-      H1_FECollection fec2(order, dim);
-      FiniteElementSpace fes2(&mesh, &fec2, dim);
-
       fes1.GetEssentialTrueDofs(ess_bdr, ess_trial_tdof_list);
-      fes2.GetEssentialTrueDofs(ess_bdr, ess_test_tdof_list);
-
-      GridFunction field(&fes1), field2(&fes2);
-
-      MixedBilinearForm gform(&fes1, &fes2);
-      gform.AddDomainIntegrator(new GradientIntegrator);
-      gform.Assemble();
-
-      // Project u = f1
-      FunctionCoefficient fcoeff1(f1);
-      field.ProjectCoefficient(fcoeff1);
-
-      VectorFunctionCoefficient fcoeff2(dim, gradf1);
-      LinearForm lf(&fes2);
-      lf.AddDomainIntegrator(new VectorDomainLFIntegrator(fcoeff2));
-      lf.Assemble();
-
-      OperatorHandle G;
-      Vector X, B;
-      gform.FormRectangularLinearSystem(ess_trial_tdof_list,
-                                        ess_test_tdof_list,
-                                        field,
-                                        lf,
-                                        G,
-                                        X,
-                                        B);
-
-      G->Mult(field, field2);
-
-      subtract(B, field2, field2);
-      REQUIRE(field2.Norml2() == MFEM_Approx(0.0));
+      GridFunction field(&fes1);
+
+      for (int i = 0; i < 3; i++)
+      {
```

Comment:
Similar to comments above, it would be better to use `GENERATE` instead of this loop